packages/web3-eth-admin/types/tests/admin-tests.ts

```diff
@@ -0,0 +1,122 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file admin-tests.ts
+ * @author Prince Sinha  <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+import {Admin} from 'web3-eth-admin';
+import {NodeInfo} from 'web3-core';
+
+const admin = new Admin('http://localhost:8545');
+
+// $ExpectType Promise<boolean>
+admin.addPeer('enode://486cbdc847079d423515a76add04f706f3b76c52120ec195b7dacbe3ad9164cf58bd389e3fc6075457b11a132d3627a625d8b14e0f0b73a86e5a3415fa4c6042@127.0.0.1:30303');
+
+// $ExpectType Promise<boolean>
+admin.addPeer(
+    'enode://486cbdc847079d423515a76add04f706f3b76c52120ec195b7dacbe3ad9164cf58bd389e3fc6075457b11a132d3627a625d8b14e0f0b73a86e5a3415fa4c6042@127.0.0.1:30303',
+    (error: Error, result: boolean) => {}
+);
+
+// $ExpectType Promise<string>
+admin.getadminDataDirectory();
+
+// $ExpectType Promise<string>
+admin.getadminDataDirectory(
+    (error: Error, result: string) => {}
+);
+
+// $ExpectType Promise<NodeInfo>
+admin.getNodeInfo();
+
+// $ExpectType Promise<NodeInfo>
+admin.getNodeInfo(
+    (error: Error, result: NodeInfo) => {}
+);
+
+// $ExpectType Promise<[]>
+admin.getPeers();
+
+// $ExpectType Promise<[]>
+admin.getPeers(
+    (error: Error, result: []) => {}
+);
+
+// $ExpectType Promise<string>
+admin.setSolc('/usr/bin/solc');
+
+// $ExpectType Promise<string>
+admin.setSolc(
+    '/usr/bin/solc',
+    (error: Error, result: string) => {}
+)
+
+// $ExpectType Promise<boolean>
+admin.startRPC();
```

Comment:
You have not tested every single possibility for `startRPC`, please can you.

Reply:
yes, I will.